### PR TITLE
[Cdn] Rename afd prefix to front door

### PR DESCRIPTION
**Pull request by Yao725**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md [hunk 1]
```diff
@@ -217,19 +217,21 @@ directive:
       for (var key in $) {
```
Comment (ArthurMa1978) [inline sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md:217]: Is this rule can be replaced by `rename-rules` ?

File: sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md [hunk 2]
```diff
@@ -250,10 +252,14 @@ directive:
...
       $.AFDOriginGroupUpdatePropertiesParameters.properties.trafficRestorationTimeToHealedOrNewEndpointsInMinutes['x-ms-client-name'] = 'trafficRestorationTimeInMinutes';
       $.AutoGeneratedDomainNameLabelScope['x-ms-enum'].name = 'DomainNameLabelScope';
-      $.AFDStateProperties.properties.deploymentStatus['x-ms-enum'].name = 'AfdDeploymentStatus';
-      $.AFDEndpointProtocols['x-ms-enum'].name = 'AfdEndpointProtocol';
+      $.AFDStateProperties.properties.deploymentStatus['x-ms-enum'].name = 'FrontDoorDeploymentStatus';
+      $.AFDStateProperties.properties.provisioningState['x-ms-enum'].name = 'FrontDoorProvisioningState';
+      $.AFDEndpointProtocols['x-ms-enum'].name = 'FrontDoorEndpointProtocol';
       $.ValidateSecretOutput.properties.status['x-ms-enum'].name = 'validationStatus';
-      $.AfdCustomDomainHttpsContent.properties.secret = {
+      $.FrontDoorCustomDomainHttpsContent.properties.certificateType['x-ms-enum'].name = 'FrontDoorCertificateType';
```
Comment (ArthurMa1978) [inline sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md:259]: if use `rename-rules` then here need to use origin name

File: sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md [hunk 3]
```diff
@@ -217,19 +217,21 @@ directive:
...
             if (key === 'AFDDomainHttpsParameters')
             {
-                const newKey = 'AfdCustomDomainHttpsContent'
+                const newKey = 'FrontDoorCustomDomainHttpsContent'
                 $[newKey] = $[key]
                 delete $[key]
             }
         }
-      $.AFDDomainUpdatePropertiesParameters.properties.tlsSettings['$ref'] = '#/definitions/AfdCustomDomainHttpsContent';
+      $.AFDDomainUpdatePropertiesParameters.properties.tlsSettings['$ref'] = '#/definitions/FrontDoorCustomDomainHttpsContent';
```
[the remark this answers is not in the document]
Comment (Yao725) [reply sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md:225]: I think so.

File: sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md [hunk 4]
```diff
@@ -217,19 +217,21 @@ directive:
...
       $.ActivatedResourceReference.properties.id['x-ms-format'] = 'arm-id';
       $.Usage.properties.id['x-ms-format'] = 'arm-id';
-      $.Route['x-ms-client-name'] = 'AfdRoute';
-      $.RuleSet['x-ms-client-name'] = 'AfdRuleSet';
-      $.Rule['x-ms-client-name'] = 'AfdRule';
-      $.SecurityPolicy['x-ms-client-name'] = 'AfdSecurityPolicy';
-      $.Secret['x-ms-client-name'] = 'AfdSecret';
+      $.AfdPurgeParameters['x-ms-client-name'] = 'FrontDoorPurgeParameters';
```
[the remark this answers is not in the document]
Comment (Yao725) [reply sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md:228]: I would prefer to use directive when we just want to rename a model. Rename-rules is a powerful configuration, but out initial intention is to rename the acronym with it, so I'd like to keep the usage within a certain range, unless we meet some cases that we can't well handle with directive.

File: sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md (hunk 1 above)
Comment (Yao725) [reply sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md]: Fixed

File: sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Yao725) [reply sdk/cdn/Azure.ResourceManager.Cdn/src/autorest.md]: Removed

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**